packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -43,9 +43,10 @@ export default function NavigationMenuSelector( {
 
 	// Avoid showing any currently active menu in the list of
 	// menus that can be selected.
-	const navigationMenusOmitCurrent = navigationMenus.filter(
-		( menu ) => ! currentMenuId || menu.id !== currentMenuId
-	);
+	const navigationMenusOmitCurrent =
+		navigationMenus?.filter(
+			( menu ) => ! currentMenuId || menu.id !== currentMenuId
+		) || [];
```

Comment:
I'm inclined to show the selected menu with a check next to it; it makes it clear what is currently selected (someone might go looking for their "header menu" not realising it's already selected) and also because removing the selected item from the dropdown causes the whole component to re-render and focus (which should go back to the Select Menu button after close) gets lost in the process.

Reply:
I've refactored to use the selected menu with a check. The semantics of a radio suit the scenario much better and avoid the re-rendered we mentioned. This is much better thank you 🙇‍♂️ 🙇‍♂️ 🙇‍♂️